.github/workflows/ci_macos.yaml

```diff
@@ -36,12 +31,16 @@ jobs:
           sudo ln -s /usr/local/opt/qt@5/mkspecs /usr/local/mkspecs
           sudo ln -s /usr/local/opt/qt@5/plugins /usr/local/plugins
 
+      - name: Setup ccache
+        uses: Chocobo1/setup-ccache-action@v1
```

Comment:
There are some minor improvements under the hood:
1. It automatically exports correct the `PATH` environment variable.
  Previously `export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"` was a mixup of macOS path and linux path and is not needed explicitly now.
2. It has better control when installing the ccache program via package managers.
  The previous action provider doesn't care much about package manager has up-to-date package indexes or not, it just do a simple install and in the worst case it might fail due to outdated package indexes. The new action provider has some customizable routines/flags to avoid that situation.
3. The new action provider let users to pass ccache configuration without writing platform specific code (shell code), although qbt project doesn't make use of it currently.

Reply:
Also note that unfortunately neither providers (in current state) can solve the caching issues observed in qbt project.
The caching issue is really a mixup of events that goes wrong together...
If anyone is interested here is the outline:
* For cache, GHA CI maintains a key-value list per repo. The key must be a unique string and value can be data uploaded from CI instance. There are no limits on the number of key-value entries.
* A key-value entry is immutable, that is, you cannot update an existing key with new value/data. 
* GHA CI free/open source plan limits cache entries to 5 GB per repo and when over-limit occurs it follows LRU (Least Recently Used) to evict entries.

So to ensure we have up-to-date cache, the cache key must be unique for every upload and therefore the key-value list experience frequent LRU evictions. Now this is would be fine when the evicted entries are the ones that are outdated. But here comes the big entry: vcpkg cache, about ~1.7GB itself and one addition of it to the k-v list will wipe out a lot of other entries. And the weird thing I don't know why is, `lukka/run-vcpkg` uses a fixed string for cache key, this is bad (for our use case) because other unique keys will let this fixed key become outdated (due to LRU) and thus vcpkg cache is evicted quickly and the next time a new CI job runs, it will rebuild the cache from the start all over again (long build time).

The best solution IMO would be making the k-v list mutable, IIRC this is what TravisCI do and it works well (or just better than other possible solutions). But for GHA CI this seems to be a design choice (originated from their underlying platform: MS azure CI) hopefully they would reconsider that decision.

Reply:
Just throwing this out there....what about using the `binary caching` option?

ref.: https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md

Reply:
> Just throwing this out there....what about using the `binary caching` option?
> 
> ref.: https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md
> 
> Could we also not make use of https://github.com/actions/cache?

It could be good if you know how to get it 100% working...